learn/shorts/functional-programming-theory.qmd

```diff
@@ -0,0 +1,202 @@
+---
+title: "A brief introduction to functional programming"
+description: |
+  Functional programming is a programming style. This
+  short introduces the basics of functional programming, its advantages,
+  and how it differs from more imperative programming (like Object Oriented Programming).
+author: "Kris Beicher"
+date: "2025-03-19"
+# date-modified: ""
+categories:
+  - python
+  - functional programming
+---
+
+## Motivation for this micro-lesson
+
+-   This lesson aims to introduce the basics of functional programming
+    to those who are new to programming or have only worked with
+    imperative programming languages. We will cover the fundamentals of
+    functional programming, its advantages, and how it differs from
+    Object-Oriented Programming. By the end of this lesson, learners
+    should be comfortable with core functional programming concepts and
+    understand why this paradigm is useful.
+
+## Assumed knowledge of reader
+
+-   Basic programming knowledge
+-   Basic Python knowledge
+-   Basic understanding of how functions work
+
+## Necessary software
+
+-   Python editor of choice
+
+## Learning goal
+
+The goal with this learning short is for the reader to understand the
+basics of functional programming and how it differs from imperative
+programming. And to be able to answer the following questions:
+
+-   What is an immutable data structure?
```

Comment:
Another item here that you talk about is: declarative vs imperative, or maybe even get into recursive functions (calling the same function within itself)